generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java

```diff
@@ -663,8 +663,52 @@ private void generateClassWriter(String className, StructSpec struct,
             Versions parentVersions) {
         headerGenerator.addImport(MessageGenerator.WRITABLE_CLASS);
         headerGenerator.addImport(MessageGenerator.OBJECT_SERIALIZATION_CACHE_CLASS);
-        buffer.printf("@Override%n");
-        buffer.printf("public void write(Writable _writable, ObjectSerializationCache _cache, short _version) {%n");
+        if (className.equals("UpdateMetadataRequestData")) {
```

Comment:
Can we add a comment to briefly state that because we only want to reuse the ByteBuffer in UpdateMetadataRequest?

Reply:
Thanks. I've added the comments.